lint: update list of spelling linter false positives, bump to codespell 2.0.0

theStack opened the pull request:
This small patch updates the ignore list for the spelling linter script (which uses `codespell`), both removing false-positives that are not relevant anymore and adding new ones. As [suggested by jonatack](https://github.com/bitcoin/bitcoin/pull/20762#issuecomment-750889701)~~, whose last name is now also part of the list :)~~. Also changed the linter script to not check the gitian keys file, as [suggested by hebasto](https://github.com/bitcoin/bitcoin/pull/20817#discussion_r550763409). The codespell version used is bumped to most recent version 2.0.0, which is more aware of some terms that were previously needed in the ignorelist for v1.17.1, see https://github.com/bitcoin/bitcoin/pull/20817#issuecomment-753428669.

Running spelling linter on master branch (repeated findings in the same file are removed to keep the output short):
```
$ ./test/lint/lint-spelling.sh
contrib/gitian-keys/keys.txt:16: Atack ==> Attack
doc/developer-notes.md:1284: inout ==> input, in out
doc/psbt.md:122: Asend ==> Ascend, as end
src/bench/verify_script.cpp:27: Keypair ==> Key pair
src/blockencodings.h:30: Unser ==> Under, unset, unsure, user
src/compressor.h:65: Unser ==> Under, unset, unsure, user
src/core_read.cpp:131: presense ==> presence
src/index/disktxpos.h:21: blockIn ==> blocking
src/net_processing.h:67: anounce ==> announce
src/netaddress.h:486: compatiblity ==> compatibility
src/primitives/transaction.h:35: nIn ==> inn, min, bin, nine
src/qt/bitcoinunits.cpp:101: nIn ==> inn, min, bin, nine
src/rpc/blockchain.cpp:2150: nIn ==> inn, min, bin, nine
src/rpc/misc.cpp:198: nIn ==> inn, min, bin, nine
src/script/bitcoinconsensus.cpp:81: nIn ==> inn, min, bin, nine
src/script/bitcoinconsensus.h:63: nIn ==> inn, min, bin, nine
src/script/interpreter.cpp:1279: nIn ==> inn, min, bin, nine
src/script/interpreter.h:222: nIn ==> inn, min, bin, nine
src/script/sign.cpp:17: nIn ==> inn, min, bin, nine
src/script/sign.h:39: nIn ==> inn, min, bin, nine
src/serialize.h:181: Unser ==> Under, unset, unsure, user
src/signet.cpp:142: nIn ==> inn, min, bin, nine
src/test/base32_tests.cpp:17: fo ==> of, for
src/test/base64_tests.cpp:17: fo ==> of, for
src/test/script_tests.cpp:1509: nIn ==> inn, min, bin, nine
src/test/sighash_tests.cpp:27: nIn ==> inn, min, bin, nine
src/test/validation_tests.cpp:78: excercise ==> exercise
src/undo.h:36: Unser ==> Under, unset, unsure, user
src/validation.cpp:1403: nIn ==> inn, min, bin, nine
src/validation.h:255: nIn ==> inn, min, bin, nine
src/wallet/wallet.cpp:1532: nIn ==> inn, min, bin, nine
src/wallet/walletdb.cpp:429: Crypted ==> Encrypted
test/functional/feature_nulldummy.py:63: unnecssary ==> unnecessary
test/functional/wallet_encryption.py:81: crypted ==> encrypted
test/functional/wallet_upgradewallet.py:36: fpr ==> for, far, fps
^ Warning: codespell identified likely spelling errors. Any false positives? Add them to the list of ignored words in test/lint/lint-spelling.ignore-words.txt
```

Running spelling linter on PR branch:
```
$ ./test/lint/lint-spelling.sh
src/core_read.cpp:131: presense ==> presence
src/net_processing.h:67: anounce ==> announce
src/netaddress.h:486: compatiblity ==> compatibility
src/test/validation_tests.cpp:78: excercise ==> exercise
src/wallet/walletdb.cpp:429: Crypted ==> Encrypted
test/functional/feature_nulldummy.py:63: unnecssary ==> unnecessary
test/functional/wallet_encryption.py:81: crypted ==> encrypted
^ Warning: codespell identified likely spelling errors. Any false positives? Add them to the list of ignored words in test/lint/lint-spelling.ignore-words.txt
```
This list of remaining findings doesn't contain false positives anymore -- the typos are fixed in PR https://github.com/bitcoin/bitcoin/pull/20762.
Happy new year! 🍾

theStack committed:
lint: update list of spelling linter false positives

jonatack reviewed (commented):
ACK 8b4e84dbcd2a08e8a6e658d7aa684705b4f0d974

```
$ test/lint/lint-spelling.sh 
src/core_read.cpp:131: presense ==> presence
src/net_processing.h:67: anounce ==> announce
src/netaddress.h:486: compatiblity ==> compatibility
src/test/validation_tests.cpp:78: excercise ==> exercise
src/wallet/walletdb.cpp:429: Crypted ==> Encrypted
test/functional/feature_nulldummy.py:63: unnecssary ==> unnecessary
test/functional/wallet_encryption.py:81: crypted ==> encrypted
```
Well done, thanks.

practicalswift commented:
ACK 8b4e84dbcd2a08e8a6e658d7aa684705b4f0d974

> Happy new year! 🍾

Happy new year @theStack!

hebasto commented on test/lint/lint-spelling.ignore-words.txt:
```diff
@@ -1,17 +1,17 @@
-unselect
-lowercased
+asend
+atack
```
Is ignoring the `contrib/gitian-keys/keys.txt` file a more reliable and future-proof solution?

theStack replied:
Agreed that it would be better to exclude files from the spelling linter that merely consists of names. Maybe a follow-up PR would be appropriate here, where not only the gitian keys file, but all other candidates are thoroughly identified and excluded?

hebasto replied:
Why a separate pr for a one-line change?

```diff
--- a/test/lint/lint-spelling.sh
+++ b/test/lint/lint-spelling.sh
@@ -15,6 +15,6 @@ if ! command -v codespell > /dev/null; then
 fi
 
 IGNORE_WORDS_FILE=test/lint/lint-spelling.ignore-words.txt
-if ! codespell --check-filenames --disable-colors --quiet-level=7 --ignore-words=${IGNORE_WORDS_FILE} $(git ls-files -- ":(exclude)build-aux/m4/" ":(exclude)contrib/seeds/*.txt" ":(exclude)depends/" ":(exclude)doc/release-notes/" ":(exclude)src/leveldb/" ":(exclude)src/crc32c/" ":(exclude)src/qt/locale/" ":(exclude)src/qt/*.qrc" ":(exclude)src/secp256k1/" ":(exclude)src/univalue/"); then
+if ! codespell --check-filenames --disable-colors --quiet-level=7 --ignore-words=${IGNORE_WORDS_FILE} $(git ls-files -- ":(exclude)build-aux/m4/" ":(exclude)contrib/seeds/*.txt" ":(exclude)depends/" ":(exclude)doc/release-notes/" ":(exclude)src/leveldb/" ":(exclude)src/crc32c/" ":(exclude)src/qt/locale/" ":(exclude)src/qt/*.qrc" ":(exclude)src/secp256k1/" ":(exclude)src/univalue/" ":(exclude)contrib/gitian-keys/keys.txt"); then
     echo "^ Warning: codespell identified likely spelling errors. Any false positives? Add them to the list of ignored words in ${IGNORE_WORDS_FILE}"
 fi
```

theStack replied:
I think this does strictly speaking more than the PR title suggests, but OTOH I don't mind adding it. (Though, for this specific case, considering jonatacks high involvement and activity in the project though, it's quite likely that his last name will appear in some other file (e.g. in doc/) and it will have to end up in the spelling linter ignorelist again :))

Anyways, added another commit. If it creates controversy for some reason, it can easily be removed from the PR without invalidating the previous ACKs.

hebasto reviewed (changes requested):
Concept ACK.

theStack committed:
lint: ignore gitian keys file for spelling linter

Co-authored-by: Hennadii Stepanov <32963518+hebasto@users.noreply.github.com>

theStack commented:
Added a commit to ignore the gitian keys file for linting, as [suggested by hebasto](https://github.com/bitcoin/bitcoin/pull/20817#discussion_r550763409), and updated the PR description accordingly.

hebasto reviewed (commented):
Tested a63b712cad90f57cc9c7a8f0436c1e4ec7e77dc7:
```
$ codespell --version
1.17.1
$ test/lint/lint-spelling.sh 
ci/lint/06_script.sh:29: keyserver ==> key server
contrib/macdeploy/gen-sdk:60: files' ==> file's
contrib/seeds/makeseeds.py:108: dedup ==> dedupe
contrib/seeds/makeseeds.py:189: dedup ==> dedupe
contrib/verify-commits/README.md:43: keyserver ==> key server
src/Makefile.am:278: OBJEXT ==> OBJECT
src/core_read.cpp:131: presense ==> presence
src/cuckoocache.h:152: copyable ==> copiable
src/net_processing.h:67: anounce ==> announce
src/netaddress.h:486: compatiblity ==> compatibility
src/support/lockedpool.h:55: copyable ==> copiable
src/support/lockedpool.h:166: copyable ==> copiable
src/test/validation_tests.cpp:78: excercise ==> exercise
src/wallet/walletdb.cpp:429: Crypted ==> Encrypted
src/zmq/zmqpublishnotifier.cpp:231: Dedup ==> Dedupe
test/functional/feature_nulldummy.py:63: unnecssary ==> unnecessary
test/functional/wallet_encryption.py:81: crypted ==> encrypted
^ Warning: codespell identified likely spelling errors. Any false positives? Add them to the list of ignored words in test/lint/lint-spelling.ignore-words.txt
```

> This list of remaining findings doesn't contain false positives anymore -- the typos are fixed in PR #20762.

Not sure about that statement.

From the Cirrus linter job [log](https://cirrus-ci.com/task/5208611494297600?command=lint#L457):
```
ci/lint/06_script.sh:29: keyserver ==> key server
contrib/macdeploy/gen-sdk:60: files' ==> file's
contrib/seeds/makeseeds.py:108: dedup ==> dedupe
contrib/seeds/makeseeds.py:189: dedup ==> dedupe
contrib/verify-commits/README.md:43: keyserver ==> key server
src/Makefile.am:278: OBJEXT ==> OBJECT
src/core_read.cpp:131: presense ==> presence
src/cuckoocache.h:152: copyable ==> copiable
src/net_processing.h:67: anounce ==> announce
src/netaddress.h:486: compatiblity ==> compatibility
src/support/lockedpool.h:55: copyable ==> copiable
src/support/lockedpool.h:166: copyable ==> copiable
src/test/validation_tests.cpp:78: excercise ==> exercise
src/wallet/walletdb.cpp:429: Crypted ==> Encrypted
src/zmq/zmqpublishnotifier.cpp:231: Dedup ==> Dedupe
test/functional/feature_nulldummy.py:63: unnecssary ==> unnecessary
test/functional/wallet_encryption.py:81: crypted ==> encrypted
^ Warning: codespell identified likely spelling errors. Any false positives? Add them to the list of ignored words in test/lint/lint-spelling.ignore-words.txt
```

theStack commented:
@hebasto: Thanks for testing! I wasn't aware that I used already codespell 2.0.0 (recently released in 23 Nov 2020), which is obviously more aware of various terms like e.g. `objext` as Makefile term and `copyable` as C++ term (see their commit https://github.com/codespell-project/codespell/commit/45b67b427489910b0ef5e4566c9b3989be3f0f33 which fixed this).
Could either bump the codespell version from 1.17.1 to 2.0.0 in an additional (first) commit or simply restore the removed exceptions for `objext`, `copyable`, `keyserver` (and add `dedup`) from the false positives wordlist change.

hebasto commented:
> Could either bump the codespell version from 1.17.1 to 2.0.0 in an additional (first) commit or simply restore the removed exceptions for `objext`, `copyable`, `keyserver` (and add `dedup`) from the false positives wordlist change.

I think the former is fine.

theStack committed:
test: bump codespell linter version to 2.0.0

hebasto commented on test/lint/lint-spelling.ignore-words.txt:
```diff
@@ -1,17 +1,16 @@
 ser
-unselect
-lowercased
+asend
+blockin
+fo
+fpr
+inout
+keypair
+nin
+unser
```
nit: The sorting mode should be lexicographical or "natural" (i.e., as reported by the linter, that is lexicographical for containing file paths), but not a mix of both.

theStack replied:
Thanks, I decided for the lexicographical order as I think it's easier to read and maintain. Done.

hebasto commented on test/lint/lint-spelling.ignore-words.txt:
```diff
@@ -1,17 +1,16 @@
+fo
+fpr
+inout
+keypair
```
nit: Maybe leave it for #20762?
s/Keypair/Key pair/ seems reasonable in https://github.com/bitcoin/bitcoin/blob/ae8f79713543d3db24a886cc27340b1570c2657d/src/bench/verify_script.cpp#L27

theStack replied:
Agreed, done.

hebasto reviewed (approved):
ACK 852cd43b3e22d84aa44ec6aa42b24dd3cf5c6c01, tested on Linux Mint 20 (x86_64).

theStack commented:
Force-pushed, addressed @hebasto's suggestions of keeping the ignore-list in order (https://github.com/bitcoin/bitcoin/pull/20817#discussion_r550874221) and removing the "keypair" exception (https://github.com/bitcoin/bitcoin/pull/20817#discussion_r550874370).

hebasto reviewed (approved):
re-ACK f3ba916e8b5b5ee2a381cef38882671eadb231df, only suggested changes since my [previous](https://github.com/bitcoin/bitcoin/pull/20817#pullrequestreview-560632881) review.

jonatack reviewed (commented):
ACK f3ba916e8b5b5ee2a381cef38882671eadb231df I don't know if there are any particular issues with bumping codespell to v2.0.0, but locally running the spelling linter and the cirrus job at https://cirrus-ci.com/task/5004066998714368 both LGTM. Thanks for also verifying and removing the unused words from the ignore list.
